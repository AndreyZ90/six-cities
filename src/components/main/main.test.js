import React from 'react';
import TestRenderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import leaflet from 'leaflet';

import Main from '@/components/main/main';

leaflet.map = () => {};

const mockStore = configureStore([]);

const offers = [
  {
    id: 1,
    isFavorite: true,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    price: 120,
    rating: 4.9,
    title: `Beautiful & luxurious studio at great location`,
    type: `apartment`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8
    },
    city: {
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      },
      name: `Amsterdam`
    }
  },
  {
    id: 2,
    isFavorite: true,
    isPremium: false,
    previewImage: `img/apartment-02.jpg`,
    price: 99,
    rating: 3.6,
    title: `Wood and stone place`,
    type: `room`,
    location: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198,
      zoom: 8
    },
    city: {
      location: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 10
      },
      name: `Amsterdam`
    }
  }
];

const cityList = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
const currentCity = `Amsterdam`;
const currentSort = `Popular`;
const activeItem = -1;

const store = mockStore({
  user: {
    email: `test@gmail.com`,
    authStatus: `AUTH`
  }
});

describe(`Main component snapshot`, () => {
  test(`Should correctly render Main component`, () => {
    const tree = TestRenderer.create(
        <Provider store={store}>
          <Router>
            <Main
              offers={offers}
              cities={cityList}
              currentCity={currentCity}
              currentSort={currentSort}
              activeItem={activeItem}
              onCurrentSortChange={() => {}}
              onActiveItemChange={() => {}}
            />
          </Router>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
