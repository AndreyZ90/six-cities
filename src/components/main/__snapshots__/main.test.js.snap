// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component snapshot Should correctly render Main component 1`] = `
<div
  className="cities"
>
  <div
    className="cities__places-container container"
  >
    <section
      className="cities__places places"
    >
      <h2
        className="visually-hidden"
      >
        Places
      </h2>
      <b
        className="places__found"
      >
        2
         places to stay in 
        Amsterdam
      </b>
      <form
        action="#"
        className="places__sorting"
        method="get"
      >
        <span
          className="places__sorting-caption"
        >
          Sort by
        </span>
        <span
          className="places__sorting-type"
          onClick={[Function]}
          tabIndex={0}
        >
          Popular
          <svg
            className="places__sorting-arrow"
            height={4}
            width={7}
          >
            <use
              xlinkHref="#icon-arrow-select"
            />
          </svg>
        </span>
        <ul
          className="places__options places__options--custom "
        >
          <a
            href="/amsterdam?sort=Popular"
            onClick={[Function]}
          >
            <li
              className="places__option places__option--active"
              onClick={[Function]}
              tabIndex={0}
            >
              Popular
            </li>
          </a>
          <a
            href="/amsterdam?sort=Price: low to high"
            onClick={[Function]}
          >
            <li
              className="places__option "
              onClick={[Function]}
              tabIndex={0}
            >
              Price: low to high
            </li>
          </a>
          <a
            href="/amsterdam?sort=Price: high to low"
            onClick={[Function]}
          >
            <li
              className="places__option "
              onClick={[Function]}
              tabIndex={0}
            >
              Price: high to low
            </li>
          </a>
          <a
            href="/amsterdam?sort=Top rated first"
            onClick={[Function]}
          >
            <li
              className="places__option "
              onClick={[Function]}
              tabIndex={0}
            >
              Top rated first
            </li>
          </a>
        </ul>
      </form>
      <div
        className="tabs__content cities__places-list places__list"
      >
        <article
          className="cities__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="place-card__image-wrapper"
          >
            <a>
              <img
                alt="Place image"
                className="place-card__image"
                height={200}
                src="img/apartment-01.jpg"
                width={260}
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  120
                </b>
                <span
                  className="place-card__price-text"
                >
                   / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button place-card__bookmark-button--active"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height={19}
                  width={18}
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/1"
                onClick={[Function]}
              >
                Beautiful & luxurious studio at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Apartment
            </p>
          </div>
        </article>
        <article
          className="cities__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="place-card__image-wrapper"
          >
            <a>
              <img
                alt="Place image"
                className="place-card__image"
                height={200}
                src="img/apartment-02.jpg"
                width={260}
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  99
                </b>
                <span
                  className="place-card__price-text"
                >
                   / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button place-card__bookmark-button--active"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height={19}
                  width={18}
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/2"
                onClick={[Function]}
              >
                Wood and stone place
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Private Room
            </p>
          </div>
        </article>
      </div>
    </section>
    <div
      className="cities__right-section"
    >
      <section
        className="cities__map map"
      />
    </div>
  </div>
</div>
`;
